File: core/internet/information_diffusion_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.internet.semantic_propagation_engine import model_semantic_propagation


def analyze_information_diffusion(seed: str, graph_edges: List[Dict[str, Any]]) -> Dict[str, Any]:
    return model_semantic_propagation(seed, graph_edges)

File: src/internet/informationDiffusionEngine.ts
/**
 * Converted from Python: core/internet/information_diffusion_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import { modelSemanticPropagation } from "./semanticPropagationEngine.js";

export function analyzeInformationDiffusion(seed: any, graph_edges: any): any {
  return modelSemanticPropagation(seed, graph_edges);
}
export { modelSemanticPropagation };
